iModelCore/ECPresentation/Source/UpdateHandler.cpp

```diff
@@ -596,6 +557,7 @@ struct HierarchyRefresher
 
         AffectedHierarchyLevelIdentifier grandparentLevel(parentHierarchyLevelIdentifier, grandParent.IsValid() ? grandParent->GetKey().get() : nullptr);
 
+        newProvider.GetContext().GetHierarchyLevelLocker().Unlock();
         std::shared_ptr<IHierarchyLevelLocker> grandParentHierarchyLock = m_nodesCache->CreateHierarchyLevelLocker(grandparentLevel.GetHierarchyLevelIdentifier(), true);

```

Comment:
Do we really need to lock here? It looks like a lock gets created by `RefreshHierarchyLevel` -> `RefreshNavNodesProvider`

Reply:
We don't need. Removed